napari/_qt/_qplugins/_qnpe2.py

```diff
@@ -0,0 +1,476 @@
+"""Plugin related functions that require Qt.
+
+Non-Qt plugin functions can be found in: `napari/plugins/_npe2.py`
+"""
+
+from __future__ import annotations
+
+import inspect
+from functools import partial
+from itertools import chain
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    List,
+    Optional,
+    Tuple,
+    Union,
+    cast,
+)
+
+from app_model import Action
+from app_model.types import SubmenuItem, ToggleRule
+from magicgui.type_map._magicgui import MagicFactory
+from magicgui.widgets import FunctionGui, Widget
+from npe2 import plugin_manager as pm
+from qtpy.QtWidgets import QWidget
+
+from napari._app_model import get_app
+from napari._app_model.constants import MenuGroup, MenuId
+from napari._app_model.injection._providers import _provide_viewer
+from napari._qt._qapp_model.injection._qproviders import _provide_window
+from napari.errors.reader_errors import MultipleReaderError
+from napari.plugins import menu_item_template, plugin_manager
+from napari.plugins._npe2 import get_widget_contribution
+from napari.utils.events import Event
+from napari.utils.translations import trans
+from napari.viewer import Viewer
+
+if TYPE_CHECKING:
+    from npe2.manifest import PluginManifest
+    from npe2.plugin_manager import PluginName
+    from npe2.types import WidgetCreator
+
+    from napari.qt import QtViewer
+
+
+# TODO: This is a separate function from `_build_samples_submenu_actions` so it
+# can be easily deleted once npe1 is no longer supported.
+def _rebuild_npe1_samples_menu() -> None:
```

Comment:
This function requires having coverage in tests. It looks easy to break, so we need to be noticed.

Reply:
I think we discussed this, npe1 is slated for removal and @DragaDoncila said she was happy to manually test. If you really want test (note I am not sure if there were any exisiting tests on main regarding this) there was discussion of a npe1 fixture but that never eventuated for the above reasons. Ping @DragaDoncila.